### ISSUE TITLE
Improvements to the NuGet UI package selections in Visual Studio - Transitive package in the projects list should not be selected by default

### ISSUE THREAD

**Issue by gordon-matt** [NuGet/Home#13940]
### NuGet Product(s) Affected

Visual Studio Package Management UI

### Current Behavior

I just had to manually update dozens of NuGet packages (mostly the latest `8.*.*` Microsoft packages) for dozens of projects in a single solution.. each package one-by-one.. VERY time consuming, and I reckon there are 2 things that can be improved to save time in these situations

### Desired Behavior

1. We didn't want to update to .NET 9 packages yet.. only the latest `8.*.*` packages. There should be a way to filter packages by specified .NET target framework, so we can bulk update all packages to whatever we want. So in our case, we're targeting .NET 8, so all latest `8.*.*` packages should show.. not any .NET 9 packages.

2. When updating a NuGet package, the default checked projects should ONLY include those where the package is top-level, not transitive. We have almost 200 projects in the solution and it's annoying when updating a NuGet package auto selects all projects whether it's top-level or transitive. Maybe a dropdown to select "all/none/top-level only" would be good here? Otherwise just don't auto select the transitive ones, please. Unless maybe transitive is all there is for the selected package.

**Comment by Pilchard123** [comment]
I agree with both of your points (in fact I found this by checking before raising them myself), though I suspect you may get better traction if they were separate issues - one for not auto-selecting transitive references, and one for filtering by target framework.

**Comment by martinrrm** [comment]
Hi! Thank you for your feedback, I agree with you on the not auto select transitive references, for number one I think it's a duplicate of https://github.com/NuGet/Home/issues/6566. Since you already have some traction in this ticket, I'll suggest editing the description to just mention the "not auto select transitive references".

**Comment by Pilchard123** [comment]
@martinrrm - I suspect you linked the wrong issue, because this _is_ 13940.

**Comment by martinrrm** [comment]
Updated the issue link

**Comment by nkolev92** [comment]
1 is https://github.com/NuGet/Home/issues/5725
We'll track 2 specifically in this issue.

**Comment by Banner-Keith** [comment]
In my case a lot of the times I have some solution with a project hierarchy like this:

Project.Mvc
Project.Console
Project.Common

If the dependency is transitive in .Mvc and .Console only because it is being referenced in .Common, it would never make sense to have update add the reference to all three when just updating .Common would resolve all.

On the other hand there may be times where the transitive dependency path is external to the solution and maybe it's transitive in .Mvc and .Console, but not the common library, then it may make sense to offer it, especially if there's a vulnerability.

**Comment by jjanuszkiewicz** [comment]
It would be great to see the transitive selection problem fixed. It makes updating packages very tedious having to unselect projects all of the time.
